File: Back/tpi-teg-grupo-11/src/main/java/ar/edu/utn/frc/tup/piii/models/Objective.java
package ar.edu.utn.frc.tup.piii.models;

import ar.edu.utn.frc.tup.piii.enums.Color;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Objective {
    private Long id;
    private String description;
    private ObjectiveType objectiveType;
    private Color color;
    private int firstRegionTerritoriesNeeded;
    private int secondRegionTerritoriesNeeded;
    private int thirdRegionTerritoriesNeeded;
    private int fourthRegionTerritoriesNeeded;
    private int fifthRegionTerritoriesNeeded;
    private int sixthRegionTerritoriesNeeded;
}

File: Front/tpi-teg-grupo-11/src/app/core/models/class/objective.ts
import {ObjectiveType} from './objective-type';
import {Color} from '../../enums/color';

export interface Objective {
  objectiveId: number;
  description: string;
  objectiveType: ObjectiveType;
  color: Color;
  firstRegionTerritoriesNeeded: number;
  secondRegionTerritoriesNeeded: number;
  thirdRegionTerritoriesNeeded: number;
  fourthRegionTerritoriesNeeded: number;
  fifthRegionTerritoriesNeeded: number;
}
